src/dev/flang/ast/Impl.java

```diff
@@ -522,13 +525,42 @@ AbstractType inferredType(Resolution res, AbstractFeature f)
     return switch (_kind)
       {
       case FieldDef    -> _initialValue.typeIfKnown();
-      case RoutineDef  -> _code.typeIfKnown();
+      case RoutineDef  -> typeFromRedefinedFeatures(f);
       case FieldActual -> typeFromInitialValues(res, f, false);
       default -> throw new Error("missing case "+_kind);
       };
   }
 
 
+  AbstractType typeFromRedefinedFeatures(AbstractFeature f)
+  {
+    AbstractType result;
+
+    if (PRECONDITIONS) require
+      (_kind == Kind.RoutineDef);
+
+    var rf = f.redefines()
+              .stream()
+              .filter(x -> x.state().atLeast(State.RESOLVED_TYPES))
+              .map(x -> x.resultType())
+              .filter(x -> !x.isGenericArgument())
+              .filter(x -> x.generics().size() == 0)
+              .filter(x -> !x.isThisType())
```

Comment:
Instead of these restrictions, it should be possible to use `AbstractFeature.handDownNonOpen` to get the type after it was inherited. Admittedly,  `handDownNonOpen` could have a nicer name `handDownType` or `inheritedType` or similar. 
The remaining restriction before using `handDownNonOpen` is that `!x.isOpenGeneric()`. Would be great if you could try to create an example that uses an open generic result type and redefines that feature to check what makes sense here (in a separate issue / PR).